getting_started/scripting/c_sharp/c_sharp_basics.rst

```diff
@@ -3,93 +3,122 @@
+A good starting point for checking its capabilities is the
+`Compatibility <http://www.mono-project.com/docs/about-mono/compatibility/>`_
+page in the Mono documentation.
 
 .. note:: This is **not** a full-scale tutorial on the C# language as a whole.
         If you aren't already familiar with its syntax or features,
-        see the `Microsoft C# guide <https://docs.microsoft.com/en-us/dotnet/csharp/index>`_ or look for a suitable introduction elsewhere.
-
-Setup C# for Godot
-------------------
-
-Download and install the `Mono <http://www.mono-project.com/download/>`_ SDK.
-
-If you are using Godot 3.0.2, you must use Mono 5.4.
-
-Godot 3.0.3+ requires Mono 5.12 on all platforms.
+        see the
+        `Microsoft C# guide <https://docs.microsoft.com/en-us/dotnet/csharp/index>`_
+        or look for a suitable introduction elsewhere.
 
-.. note:: To download Mono v5.12+ on a Mac, locate the "Stable Channel" link from the `Mono Downloads Page <http://www.mono-project.com/download/>`_. The Visual Studio channel is an earlier version of Mono and will not work with Godot 3.0.3+.
+Setting up C# for Godot
+-----------------------
 
-You also need MSBuild (at least version 15.0), which should come with the Mono installation.
+Download and install the latest version of the
+`Mono SDK <http://www.mono-project.com/download/>`_. As of Godot 3.1 beta 3,
+the version number doesn't matter since Godot bundles its own Mono 5.18
+installation. We only need the Mono installation for NuGet and MSBuild
+which are required to use C# in Godot.
 
-.. note:: For instructions on installing older versions of Mono on Linux, see `this page <http://www.mono-project.com/docs/getting-started/install/linux/#accessing-older-releases>`_.
-        Older versions of Mono for macOS and Windows can be found `here <https://download.mono-project.com/archive/>`_.
+.. note:: To download Mono on macOS, use the "Stable Channel" link
+          from the `Mono Downloads Page <http://www.mono-project.com/download/>`_.
+          The Visual Studio channel is an earlier version of Mono and
+          will not work.
 
-Additionally, your Godot version must have Mono support enabled, so ensure you download the **Mono version** of Godot.
+Additionally, your Godot version must have Mono support enabled,
+so make sure to download the **Mono version** of Godot.
+If you are building Godot from source, make sure to follow the steps to
+enable Mono support in your build as outlined in the
+:ref:`doc_compiling_with_mono` page.
 
-If you are building Godot from source, make sure to follow the steps to include Mono support in your build outlined on the :ref:`doc_compiling_with_mono` page.
-
-In summary, you must have installed 1) the correct Mono SDK version for your Godot version, and 2) The Mono version of Godot.
+In summary, you must have installed the correct Mono SDK version for your
+Godot version **and** the Mono-enabled version of Godot.
```

Comment:
This contradicts what is said previously:

>As of Godot 3.1 beta 3,
the version number doesn't matter since Godot bundles its own Mono 5.18
installation.